src/style/style.js

```diff
@@ -847,7 +867,9 @@ class Style extends Evented {
             glyphs: this.stylesheet.glyphs,
             transition: this.stylesheet.transition,
             sources: mapObject(this.sourceCaches, (source) => source.serialize()),
-            layers: this._order.map((id) => this._layers[id].serialize())
+            layers: this._order.map((id) => this._layers[id])
+                .filter((layer) => layer.type !== 'custom')
+                .map((layer) => layer.serialize())
```

Comment:
This is duplicated with `_serializeLayers`. Let's DRY up. Maybe also write with a `for` loop / `push` for less short-lived allocations?